x/ref/lib/vdl/codegen/vdlgen/const.go

```diff
@@ -129,6 +130,92 @@ func UntypedConst(v *vdl.Value, pkgPath string, imports codegen.Imports) string
 	}
 }
 
+// JSON returns a JSON representation of a value. This was adapted from
```

Comment:
for some reason I thought there was an easier way to convert vom to json without stub generation? Maybe ping Todd? It's fine to submit this anyway.

Reply:
@tatatodd: is there a simple better way to do this?

Reply:
Hi guys!

Yeah, we had design docs describing how to convert directly from the VOM serialization format to JSON, without materializing the data into an in-memory vdl.Value.  But as-far-as-I-remember, we never got around to implementing that.  I browsed some of the code, and don't see hints of it anywhere either.

Note that Razvan's approach doesn't require stub generation.  Instead he's relying on the two-phase conversion:
1) serialized VOM -> in memory vdl.Value conversion
2) in memory vdl.Value -> JSON

This is a fine approach; step (1) is already implemented, in this PR he's implementing step (2).

My only nit for the code is the location.  The `v.io/x/ref/lib/vdl/codegen` directory is meant to hold the code generators for various languages.  E.g. we have code generators for golang, java, javascript, swift and vdl.  It's weird to have the generator for JSON to live within the `v.io/x/ref/lib/vdl/codegen/vdlgen` package, since that's meant for codegen to the VDL language, and JSON is not VDL (not even a subset).

So in my mind it this functionality belongs in the package `v.io/x/ref/lib/vdl/codegen/json`, and the function should be called something like `Const`, so the user invokes it via `json.Const` (and to match similar functions in the codegen packages for other languages).  And ideally the `javascript` codegen, that this implementation was based on, would just use this directly.

Hope this helps!  :)

Reply:
Thanks for the comment @tatatodd! The move to `v.io/x/ref/lib/vdl/codegen/json` should be easy. In other languages I see a `Generate` function and not a `Const` one ([1], [2]). The fact that the `javascript` should use is also make sense. I'll look into all addressing all these. :-)

[1] https://godoc.org/v.io/x/ref/lib/vdl/codegen/golang#Generate
[2] https://godoc.org/v.io/x/ref/lib/vdl/codegen/javascript#Generate